### PR TITLE
C#: Add output assembly to compilation

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR stores the output assembly in the `compilation`. 

- [x] Stats https://jenkins.internal.semmle.com/job/Stats/job/CSharp/27/
- [x] DB upgrade folder
- [x] Differences job https://jenkins.internal.semmle.com/job/Changes/job/CSharp-Differences/757/

### CODE REVIEW

File: csharp/ql/src/semmlecode.csharp.dbscheme [hunk 1]
```diff
@@ -7,11 +7,13 @@
...
  * The `id` simply identifies the invocation, while `cwd` is the working
- * directory from which the compiler was invoked.
+ * directory from which the compiler was invoked. The `assembly` identifies
+ * the output assembly of the compilation.
  */
 compilations(
     unique int id : @compilation,
-    string cwd : string ref
+    string cwd : string ref,
+    int assembly: @assembly ref
```
Comment (yo-h) [inline csharp/ql/src/semmlecode.csharp.dbscheme:16]: The `compilations` relation is shared between multiple languages (currently C++, C# and Java). In the interest of maintaining compatibility between languages, have you considered implementing the additional information as a separate relation? In Java, for example, a compilation typically results in multiple `.class` files, not a single assembly.

Comment (tamasvajk) [reply csharp/ql/src/semmlecode.csharp.dbscheme:16]: I have no preference. I'll change it to a separate relation. How is the `compilations` shared between languages?

File: csharp/ql/src/semmle/code/csharp/Location.qll [hunk 2]
```diff
@@ -171,6 +172,9 @@ class Assembly extends Location, Attributable, @assembly {
   /** Gets the version of this assembly. */
   Version getVersion() { assemblies(this, _, _, _, result) }
 
+  /** Gets the compilation producing this assembly if any. */
```
Comment (hvitved) [inline csharp/ql/src/semmle/code/csharp/Location.qll:175]: comma after `assembly`

**Review by hvitved — CHANGES_REQUESTED** [review]
LGTM, one trivial comment.

**Review by hvitved — APPROVED** [review]

File: csharp/ql/src/semmlecode.csharp.dbscheme (hunk 1 above)
Comment (yo-h) [reply csharp/ql/src/semmlecode.csharp.dbscheme]: Manual sync between dbscheme files at this time. :see_no_evil:

**Pull request merged**